File: 백준/Silver/10845. 큐/큐.py
# queue

import sys
input = sys.stdin.readline().rstrip()
N = int(input)

slist = []

def push(X) : 
    slist.append (X)
    
def pop() :
    if slist == [] :
        print (-1)
        
    else :
        print (slist[0])
        slist.pop(0)
        
def size() :
    print (len(slist))
    
def empty() :
    if slist == [] :
        print (1)
        
    else :
        print (0)
        
def front() :
    if slist == [] :
        print (-1)
        
    else :
        print (slist[0])

def back() :
    if slist == [] :
        print (-1)
        
    else :
        print (slist[-1])

for i in range (N) :
    input = sys.stdin.readline().rstrip()
    
    if 'push' in input : 
        push(input.split(' ')[1])
        
    elif input == 'pop' :
        pop()
        
    elif input == 'size' :
        size()
        
    elif input == 'empty' : 
        empty()
        
    elif input == 'front' :
        front()
        
    elif input == 'back' : 
        back()

File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <cstdio>
#include <queue>

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr);
    
    std::queue<int> queue;
    long long int N;
    long long int counting = 0;
    
    std::cin >> N;
    
    for (int i = 0; i < N; i++) {
        std::string command;
        std::cin >> command;

        if (command.compare("push") == 0) {
            long long int X;
            std::cin >> X;
            
            queue.push(X);
            counting += 1;
        }
        
        if (command.compare("pop") == 0) {
            if (counting > 0) {
                std::cout << queue.front() << "\n";
                queue.pop();
                counting -= 1;
            }
            
            else {
                std::cout << -1 << "\n";
            }
        }
        
        if (command.compare("size") == 0) {
            std::cout << counting << "\n";
        }
        
        if (command.compare("empty") == 0) {
            if (counting == 0) {
                std::cout << 1 << "\n";
            }
            
            else {
                std::cout << 0 << "\n";
            }
        }
        
        if (command.compare("front") == 0) {
            if (counting == 0) {
                std::cout << -1 << "\n";
            }
            
            else {
                std::cout << queue.front() << "\n";
            }
        }
        
        if (command.compare("back") == 0) {
            if (counting == 0) {
                std::cout << -1 << "\n";
            }
            
            else {
                std::cout << queue.back() << "\n";
            }
        }
    }
    
    return 0;
}